torch/_refs/__init__.py

```diff
@@ -1822,6 +1824,73 @@ def tensor_split(
         return tuple(splits)
 
 
+def hsplit(
+    a: TensorLikeType, indices_or_sections: DimsType
+) -> Tuple[TensorLikeType, ...]:
+    if a.ndim < 1:
+        msg = (
+            "torch.hsplit requires a tensor with at least 1 dimension, but got a tensor with "
+            + str(a.ndim)
+            + " dimensions!"
+        )
+        raise RuntimeError(msg)
+    dim = 0 if a.ndim == 1 else 1
+    if isinstance(indices_or_sections, int):
+        split_size = indices_or_sections
+        if not (split_size != 0 and a.shape[dim] % split_size == 0):
```

Comment:
double negation here is a little hard to read -- maybe propagate the `not`?

Reply:
is this error message redundant with error checking done by tensor_split?

Reply:
I think this msg is better than the one that `tensor_split` throws

`tensor_split: number of sections must be greater than 0, but was 0` vs
`torch.hsplit attempted to split along dimension 1, but the size of the dimension 5 is not divisible by the split_size 0!`

Reply:
OK